cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -196,6 +198,18 @@ func validateAutoSpcMaxPool(spc *apis.StoragePoolClaim) error {
 	return nil
 }
 
+// validateBDCount validates the block device count in case of manual
+// provisioning
+func validateBDCount(spc *apis.StoragePoolClaim) error {
+	if isManualProvisioning(spc) {
+		minReqBDCount := blockdevice.DefaultDiskCount[spc.Spec.PoolSpec.PoolType]
+		if len(spc.Spec.BlockDevices.BlockDeviceList)%minReqBDCount != 0 {
+			return errors.Errorf("validation of spc object is failed as invalid no.of of block device  in spc %s", spc.Name)
+		}
+	}
+	return nil
+}
+
```

Comment:
Consider a case where one want to provision a `mirrored` cStor pool and has provided only 2 BD entries on SPC.
Also -- those 2 BDs will belong from 2 different nodes. Then in this case the validation should fail.
Can the function `validateBDCount` handle above validation ?

Reply:
Putting a UT around this would help.

Reply:
Consider this is like a sanity check  and above scenario is covered in `selectNode`

Reply:
then, this is not required